miriway.cpp

```diff
@@ -203,6 +205,116 @@ struct ShellPids
         }
     }
 };
+
+class ShellComponentLaunchManager
+{
+public:
+    ShellComponentLaunchManager(ShellPids& shell_pids, ExternalClientLauncher& client_launcher, MirRunner& runner)
```

Comment:
The multiple dependencies (on `ShellPids`, `ExternalClientLauncher` and `MirRunner`) make me wonder about separation of concerns. This feels overly complex. (Are you trying to nerdsnipe me again?)

Reply:
Ok I can fix XD I don't _think_ it needs the dependencies, probably just another callback function

Reply:
Updated :+1: